iis-common/src/main/java/eu/dnetlib/iis/common/cache/CacheStorageUtils.java

```diff
@@ -0,0 +1,203 @@
+package eu.dnetlib.iis.common.cache;
+
+import java.io.File;
+
+import org.apache.avro.generic.GenericRecord;
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+
+import com.beust.jcommander.Parameter;
+import com.google.common.base.Preconditions;
+
+import eu.dnetlib.iis.audit.schemas.Fault;
+import eu.dnetlib.iis.common.lock.LockManager;
+import eu.dnetlib.iis.metadataextraction.schemas.DocumentText;
+import pl.edu.icm.sparkutils.avro.SparkAvroLoader;
+import pl.edu.icm.sparkutils.avro.SparkAvroSaver;
+
+
+/**
+ * Shared methods for managing cache storage.
+ * @author mhorst
+ *
+ */
+public class CacheStorageUtils {
+    
+    private CacheStorageUtils() {}
+
+    // ---------------------------- LOGIC ---------------------------------
+    
+    /**
+     * Removes trailing file separator from the path whenever defined.
+     */
+    public static String normalizePath(String cacheRootDir) {
+        Preconditions.checkArgument(StringUtils.isNotBlank(cacheRootDir), "cache root directory cannot be blank!");
+
+        if (File.separatorChar == cacheRootDir.charAt(cacheRootDir.length() - 1)) {
+            return cacheRootDir.substring(0, cacheRootDir.length() - 1);
+        } else {
+            return cacheRootDir;
+        }
+    }
+    
+    /**
+     * Builts cache path for given cache coordinates, none of the parameters can be null.
+     */
+    public static String getCacheLocation(String cacheRootDir, String cacheId, CacheRecordType cacheRecordType) {
+        Preconditions.checkArgument(StringUtils.isNotBlank(cacheRootDir), "cache root directory cannot be blank!");
+        Preconditions.checkArgument(StringUtils.isNotBlank(cacheId), "cache id cannot be blank!");
+        
+        return cacheRootDir + '/' + cacheId + '/' + cacheRecordType.name();
+    }
```

Comment:
Well, I can agree, at least partially. It would be cleaner to rely on `Path` objects but on the other hand both `AvroLoader#loadJavaRDD()` and `AvroSaver#saveJavaRDD()` rely on string parameters when dealing with HDFS locations. So after wrapping the path with `Path` object we would need to unbox it in a subsequent step. Not sure this is worth an effort...

Reply:
Ok, the cleaner code the better:) I have introduced `Path` related changes in mentioned places and in the referenced code within altered methods.

Reply:
I have followed the refactoring slope and finished at TARA caching modules. So once we merge those changes with IIS master branch you should probably rebase your branch with #1117 TARA modifications.

Reply:
Great, I'll include the changes in TARA related changes.